pandas/core/indexes/datetimes.py

```diff
@@ -464,6 +464,11 @@ def _convert_for_op(self, value):
             return _to_M8(value)
         raise ValueError("Passed item and index have different timezone")
 
+    def difference(self, other, sort=None):
```

Comment:
Can you add a docstring here? (I think you can use one of the decorators to append `Index.difference.__doc__`)

(and same below)

Reply:
added.